### PR TITLE
Address some of the concerns in #623

### PR DESCRIPTION
**Pull request by rht**
Consider this to be the lightweight version of #625. The remaining problem is the `place_agent()`, `position_agent()` API.

### CODE REVIEW

File: examples/forest_fire/forest_fire/model.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __init__(self, height=100, width=100, density=0.65):
                 # Set all trees in the first column on fire.
                 if x == 0:
                     new_tree.condition = "On Fire"
-                self.grid._place_agent((x, y), new_tree)
+                self.grid._place_agent(new_tree, (x, y))
```
Comment (rht) [inline examples/forest_fire/forest_fire/model.py]: This example is problematic because it is using an internal method.

**Comment by Corvince** [comment]
Thanks for the PR. I would suggest also changing the problematic example

> The remaining problem is the `place_agent()`, `position_agent()` API.

I also recently stumpled over `iter_neighbors()` and `neighbor_iter()` which seem to do the same thing, except the latter is an older, limited version of the former.

**Comment by rht** [comment]
Rebased.
> I would suggest also changing the problematic example

I will do this in a separate PR.

**Comment by rht** [comment]
Superseded by #1240.

**Pull request closed without merging**